### ISSUE TITLE
projection.fitWidth, projection.fitHeight?

### ISSUE THREAD

**Issue by mbostock** [d3/d3-geo#72]
Like we have *projection*.fitSize and *projection*.fitExtent, it might be nice to have a variant that only bounds a single dimension, and returns the implied value for the other dimension. For example, if you pass it the geometry of California and say *projection*.fitWidth(960) it would return the required *height* to fix California in a 960×*height* rectangle.

**Comment by mfogel** [comment]
I have a need for a method that behaves exactly like `projection.fitSize()` but only constrains one dimension. Is that the idea here? Two apis come to mind to allow this:

Option 1: allow `fitSize()` to accept null values to mean "I don't care about this dimension". Ex:

```javascript
projection = d3.geoIdentity().fitSize([960, null], object) // set the width to 960, height to fit
projection = d3.geoIdentity().fitSize([null, 960], object) // set the height to 960, width to fit
```

Option 2: add two small methods `fitWidth()` and `fitHeight()` to the api:

```javascript
projection = d3.geoIdentity().fitWidth(960, object) // set the width to 960, height to fit
projection = d3.geoIdentity().fitHeight(960, object) // set the height to 960, width to fit
```

@mbostock thoughts? Either of these look acceptable? If so I can set up a PR with an implementation.

**Comment by mbostock** [comment]
Yes, this proposal is for option 2.

**Comment by mbostock** [comment]
Fixed in #120.
